skimage/registration/_optical_flow.py

```diff
@@ -217,3 +218,152 @@ def optical_flow_tvl1(reference_image, moving_image,
+    # is the solution of the ndim x ndim linear system
+    # A[i, j] * X = b[i, j]
+    A = np.zeros(reference_image.shape + (ndim, ndim), dtype=dtype)
+    b = np.zeros(reference_image.shape + (ndim, ), dtype=dtype)
+
+    grid = np.meshgrid(*[np.arange(n, dtype=dtype)
+                         for n in reference_image.shape],
+                       indexing='ij', sparse=True)
+
+    for _ in range(num_warp):
+        if prefilter:
+            flow = ndi.filters.median_filter(flow, (1, ) + ndim * (3, ))
+
+        moving_image_warp = warp(moving_image, get_warp_points(grid, flow),
+                                 mode='nearest')
+        grad = np.stack(np.gradient(moving_image_warp), axis=0)
+        It = (grad * flow).sum(axis=0) + reference_image - moving_image_warp
+
+        # Local linear systems creation
+        for i, j in combinations_with_replacement(range(ndim), 2):
+            A[..., i, j] = A[..., j, i] = filter_func(grad[i] * grad[j])
+
+        for i in range(ndim):
+            b[..., i] = filter_func(grad[i] * It)
+
+        # Don't consider badly conditioned linear systems
+        idx = abs(np.linalg.det(A)) < 1e-14
+        A[idx] = np.eye(ndim, dtype=dtype)
+        b[idx] = 0
+
+        # Solve the local linear systems
+        flow = np.transpose(np.linalg.solve(A, b),
+                            (ndim, ) + tuple(range(ndim)))
+
+    return flow
+
+
+def optical_flow_ilk(reference_image, moving_image, *,
+                     radius=7, num_warp=10, gaussian=False,
+                     prefilter=False, dtype=np.float32):
+    """Coarse to fine optical flow estimator.
+
+    The iterative Lucas-Kanade (iLK) solver is applied at each level
+    of the image pyramid. iLK is a fast and robust algorithm
+    developped by Le Besnerais and Champagnat [4]_ and improved in
+    [5]_.
+
+    Parameters
+    ----------
+    reference_image : ndarray, shape (M, N[, P[, ...]])
+        The first gray scale image of the sequence.
+    moving_image : ndarray, shape (M, N[, P[, ...]])
+        The second gray scale image of the sequence.
+    radius : int, optional
+        Radius of the window considered around each pixel.
+    num_warp : int, optional
+        Number of times moving_image is warped.
+    gaussian : bool, optional
+        If True, a Gaussian kernel is used for the local
+        integration. Otherwise, a uniform kernel is used.
+    prefilter : bool, optional
+        whether to prefilter the estimated optical flow before each
+        image warp. This helps to remove potential outliers.
+    dtype : dtype, optional
+        Output data type: must be floating point. Single precision
+        provides good results and saves memory usage and computation
+        time compared to double precision.
+
+    Returns
+    -------
+    flow : ndarray, shape ((reference_image.ndim, M, N[, P[, ...]])
+        The estimated optical flow components for each axis.
+
+    Notes
+    -----
+    Color images are not supported.
```

Comment:
Can you specify either here or in the description above that the algorithm being implemented corresponds to Table 2 of [1]_?